test/test-robots.cpp

```diff
@@ -162,6 +162,16 @@ TEST(RobotsTest, SkipMalformedLine)
     EXPECT_TRUE(robot.allowed("/no/colon/in/this/line", "agent"));
 }
 
+TEST(RobotsTest, DoubleLeadingSlash)
```

Comment:
I'm fine with this more top-level integration-style test, but since you've modified `Agent`, I would expect a more specific unit test in `test-agent.cpp` as well.

You might also want to specifically point out that this drops support for schemeless URLs in `robots.txt`. I would give a specific example that demonstrates their failure now. I also think it would be worthwhile to add a test for more than two leading slashes in the agent. I'd also like to see some tests for exactly two and three slashes with no other paht.